### PR TITLE
sync client-go

### PR DESCRIPTION
**Pull request by deads2k**
Straight mechanical sync of client-go after updating apimachinery.

@caesarxuchao @sttts @pwittrock

I tagged it since its straight mechanical.

### CODE REVIEW

File: staging/src/k8s.io/client-go/OWNERS [hunk 1]
```diff
@@ -1,44 +0,0 @@
-approvers:
```
Comment (sttts) [inline staging/src/k8s.io/client-go/OWNERS]: intentional?

File: staging/src/k8s.io/client-go/examples/third-party-resources/main.go [hunk 2]
```diff
@@ -1,170 +0,0 @@
-/*
```
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/third-party-resources/main.go]: why are examples removed?

Comment (deads2k) [inline staging/src/k8s.io/client-go/examples/third-party-resources/main.go]: > why are examples removed?

Just ran a sync.  I guess the script isn't right anymore.  Miss an update maybe?

File: staging/src/k8s.io/client-go/OWNERS (hunk 1 above)
Comment (deads2k) [inline staging/src/k8s.io/client-go/OWNERS]: either scripts is off or someone changed the wrong file.

File: staging/src/k8s.io/client-go/examples/third-party-resources/main.go (hunk 2 above)
Comment (sttts) [inline staging/src/k8s.io/client-go/examples/third-party-resources/main.go]: https://github.com/kubernetes/kubernetes/pull/41321

File: staging/src/k8s.io/client-go/OWNERS (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/client-go/OWNERS]: https://github.com/kubernetes/kubernetes/pull/41321

**Comment by deads2k** [comment]
@sttts this looks better.

**Comment by caesarxuchao** [comment]
Thanks. The robot is disabled currently, so nothing will be published to client-go.

**Comment by deads2k** [comment]
> Thanks. The robot is disabled currently, so nothing will be published to client-go.

Is there something we can do manually, to push through once this merges?

**Comment by caesarxuchao** [comment]
Is it necessary to do another manual push to client-go? All client-go's branches are working currently, so I prefer wait for the robot to be back online.

**Comment by deads2k** [comment]
@k8s-bot unit test this: issue #39471

**Comment by deads2k** [comment]
> Is it necessary to do another manual push to client-go? All client-go's branches are working currently, so I prefer waiting for the robot to be back online.

Meaning you want this pull to wait?  If so, I'm ok with waiting another day.  You can add `do-not-merge` if you like.  Be sure to update whether or not the sync worked overnight though.  I don't want to wait too long.

**Comment by caesarxuchao** [comment]
This pull is fine, but I don't want to create another manual pull to client-go.

**Pull request merged**